hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/LeaseManager.java

```diff
@@ -92,21 +91,15 @@
   private long lastHolderUpdateTime;
   private String internalLeaseHolder;
 
+  //
   // Used for handling lock-leases
   // Mapping: leaseHolder -> Lease
-  private final SortedMap<String, Lease> leases = new TreeMap<>();
-  // Set of: Lease
-  private final NavigableSet<Lease> sortedLeases = new TreeSet<>(
-      new Comparator<Lease>() {
-        @Override
-        public int compare(Lease o1, Lease o2) {
-          if (o1.getLastUpdate() != o2.getLastUpdate()) {
-            return Long.signum(o1.getLastUpdate() - o2.getLastUpdate());
-          } else {
-            return o1.holder.compareTo(o2.holder);
-          }
-        }
-  });
+  // TreeMap has O(log(n)) complexity but it is more space efficient
+  //             compared to HashMap. Therefore, replacing TreeMap with a
+  //             HashMap can be considered to get faster O(1) time complexity
+  //             on the expense of 30% memory waste.
+  //
```

Comment:
This explanation belongs in the Jira, not in a comment in the code.  When looking at current code, it's not really clear why you are talking about TreeMap at all.

Also, I think this comment misses the point of the TreeMap.  The reason a TreeMap was used here was to maintain a sorted order, which allowed the checkLeases() to exit the while loop as soon as it hit an unexpired lease. 

The new design removes the need for the TreeMap by pruning the list it passes to checkLeases().